External/FEXCore/Source/Common/JitSymbols.cpp

```diff
@@ -1,64 +1,91 @@
 #include "Common/JitSymbols.h"
 
+#include <fcntl.h>
 #include <string>
 #include <unistd.h>
 
 #include <fmt/format.h>
 
 namespace FEXCore {
-  JITSymbols::JITSymbols() : fp{nullptr, std::fclose} {
+  JITSymbols::JITSymbols() {
   }
 
-  JITSymbols::~JITSymbols() = default;
+  JITSymbols::~JITSymbols() {
+    if (fd != -1) {
+      close(fd);
+    }
+  }
 
   void JITSymbols::InitFile() {
+    // We can't use FILE here since we must be robust against forking processes closing our FD from under us.
     const auto PerfMap = fmt::format("/tmp/perf-{}.map", getpid());
 
-    fp.reset(fopen(PerfMap.c_str(), "wb"));
-    if (fp) {
-      // Disable buffering on this file
-      setvbuf(fp.get(), nullptr, _IONBF, 0);
-    }
+    fd = open(PerfMap.c_str(), O_CREAT | O_EXCL | O_WRONLY | O_DIRECT, 0644);
   }
 
   void JITSymbols::Register(const void *HostAddr, uint64_t GuestAddr, uint32_t CodeSize) {
-    if (!fp) return;
+    if (fd == -1) return;
+    char TmpBuffer[512];
```

Comment:
Symbol names (particularly C++ ones) can easily become larger than this, can't they? Is this path critical enough to warrant not using `std::string` instead?

Reply:
Not sure of the performance implications of these yet. Switched over to std::string usage.